File: tools/compress-august-images.py
#!/usr/bin/env python3
"""
Compress JPG images in the August retreat folder
Uses PIL/Pillow which we already have installed
"""
import os
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("Error: PIL/Pillow not installed")
    print("Install with: pip install pillow (in your venv)")
    sys.exit(1)

def compress_image(input_path, output_path, quality=80):
    """Compress a JPG image"""
    try:
        img = Image.open(input_path)
        
        # Convert to RGB if needed (removes alpha channel)
        if img.mode in ('RGBA', 'LA', 'P'):
            rgb_img = Image.new('RGB', img.size, (255, 255, 255))
            if img.mode == 'P':
                img = img.convert('RGBA')
            rgb_img.paste(img, mask=img.split()[-1] if img.mode in ('RGBA', 'LA') else None)
            img = rgb_img
        elif img.mode != 'RGB':
            img = img.convert('RGB')
        
        # Save with compression
        img.save(output_path, 'JPEG', quality=quality, optimize=True, progressive=True)
        
        original_size = os.path.getsize(input_path)
        new_size = os.path.getsize(output_path)
        saved = ((original_size - new_size) / original_size * 100)
        saved_mb = (original_size - new_size) / (1024 * 1024)
        
        print(f"✓ {os.path.basename(input_path)}: {original_size / 1024 / 1024:.2f}MB → {new_size / 1024 / 1024:.2f}MB ({saved:.1f}% saved, {saved_mb:.2f}MB)")
        return True, original_size, new_size
    except Exception as e:
        print(f"✗ Error compressing {input_path}: {e}")
        return False, 0, 0

def main():
    script_dir = Path(__file__).parent
    august_dir = script_dir.parent / 'public' / 'images' / 'august retreat'
    
    if not august_dir.exists():
        print(f"Error: Directory not found: {august_dir}")
        return
    
    jpg_files = list(august_dir.glob('*.jpg')) + list(august_dir.glob('*.JPG'))
    
    if not jpg_files:
        print("No JPG files found to compress.")
        return
    
    print(f"Compressing {len(jpg_files)} JPG file(s) in August retreat folder...\n")
    
    total_original = 0
    total_new = 0
    compressed = 0
    
    for jpg_file in sorted(jpg_files):
        temp_file = jpg_file.with_suffix('.jpg.tmp')
        
        success, orig_size, new_size = compress_image(jpg_file, temp_file, quality=80)
        
        if success:
            # Replace original with compressed version
            jpg_file.unlink()
            temp_file.rename(jpg_file)
            total_original += orig_size
            total_new += new_size
            compressed += 1
    
    if compressed > 0:
        total_saved = ((total_original - total_new) / total_original * 100)
        total_saved_mb = (total_original - total_new) / (1024 * 1024)
        
        print(f"\n✓ Compression complete!")
        print(f"  Compressed: {compressed} file(s)")
        print(f"  Total size: {total_original / 1024 / 1024:.2f}MB → {total_new / 1024 / 1024:.2f}MB")
        print(f"  Saved: {total_saved:.1f}% ({total_saved_mb:.2f}MB)")
        print(f"\nImages will now load much faster! 🚀")

if __name__ == '__main__':
    main()

File: tools/compress-august-images.js
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const augustRetreatDir = path.join(__dirname, '../public/images/august retreat');

async function compressImage(inputPath, outputPath) {
  try {
    const stats = await fs.promises.stat(inputPath);
    const originalSize = stats.size;
    
    // Compress JPEG with good quality but smaller file size
    await sharp(inputPath)
      .jpeg({ 
        quality: 80,  // Good balance between quality and file size
        mozjpeg: true, // Better compression
        progressive: true // Progressive JPEG for faster perceived loading
      })
      .toFile(outputPath);
    
    const newStats = await fs.promises.stat(outputPath);
    const newSize = newStats.size;
    const saved = ((originalSize - newSize) / originalSize * 100).toFixed(1);
    const savedMB = ((originalSize - newSize) / (1024 * 1024)).toFixed(2);
    
    console.log(`✓ ${path.basename(inputPath)}: ${(originalSize / 1024 / 1024).toFixed(2)}MB → ${(newSize / 1024 / 1024).toFixed(2)}MB (${saved}% saved, ${savedMB}MB)`);
    return true;
  } catch (error) {
    console.error(`✗ Error compressing ${inputPath}:`, error.message);
    return false;
  }
}

async function main() {
  console.log('Starting image compression for August retreat...\n');
  
  try {
    const files = await fs.promises.readdir(augustRetreatDir);
    const jpgFiles = files.filter(file => 
      file.toLowerCase().endsWith('.jpg') || file.toLowerCase().endsWith('.jpeg')
    );
    
    if (jpgFiles.length === 0) {
      console.log('No JPG files found to compress.');
      return;
    }
    
    console.log(`Found ${jpgFiles.length} JPG file(s) to compress:\n`);
    
    let totalOriginalSize = 0;
    let totalNewSize = 0;
    let compressed = 0;
    
    for (const jpgFile of jpgFiles) {
      const inputPath = path.join(augustRetreatDir, jpgFile);
      const tempPath = path.join(augustRetreatDir, `${jpgFile}.tmp`);
      
      const stats = await fs.promises.stat(inputPath);
      totalOriginalSize += stats.size;
      
      if (await compressImage(inputPath, tempPath)) {
        // Replace original with compressed version
        await fs.promises.rename(tempPath, inputPath);
        const newStats = await fs.promises.stat(inputPath);
        totalNewSize += newStats.size;
        compressed++;
      } else {
        // If compression failed, remove temp file
        try {
          await fs.promises.unlink(tempPath);
        } catch {}
      }
    }
    
    const totalSaved = ((totalOriginalSize - totalNewSize) / totalOriginalSize * 100).toFixed(1);
    const totalSavedMB = ((totalOriginalSize - totalNewSize) / (1024 * 1024)).toFixed(2);
    
    console.log(`\n✓ Compression complete!`);
    console.log(`  Compressed: ${compressed} file(s)`);
    console.log(`  Total size: ${(totalOriginalSize / 1024 / 1024).toFixed(2)}MB → ${(totalNewSize / 1024 / 1024).toFixed(2)}MB`);
    console.log(`  Saved: ${totalSaved}% (${totalSavedMB}MB)`);
    
  } catch (error) {
    console.error('Error:', error.message);
    if (error.message.includes('sharp')) {
      console.error('\n⚠️  Sharp module issue. Try:');
      console.error('   npm install --platform=linux --arch=x64 sharp');
      console.error('   Or use the Python compression script instead.');
    }
  }
}

main();

